game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AaDefenseCombatValue.java

```diff
@@ -53,7 +53,13 @@ public RollCalculator getRoll() {
 
   @Override
   public StrengthCalculator getStrength() {
-    return new AaDefenseStrength(gameData, supportFromFriends, supportFromEnemies);
+    return new AaDefenseStrength(this, supportFromFriends, supportFromEnemies);
+  }
+
+  @Override
+  public int getDiceSides(final Unit unit) {
+    final int diceSides = unit.getUnitAttachment().getAttackAaMaxDieSides();
```

Comment:
This logic is copied from `TotalPowerAndTotalRolls`.  Oddly enough, `getAttackAaMaxDieSides` also checks if it is less than 0 and if so, it returns the `gameData.getDiceSides()`.